src/analyze/analyze-time-data.c

```diff
@@ -215,22 +217,43 @@ int pretty_boot_time(sd_bus *bus, char **ret) {
         return 0;
 }
 
+void unit_times_clear(UnitTimes *t) {
+        if (!t)
+                return;
+
+        FOREACH_ARRAY(d, t->deps, ELEMENTSOF(t->deps))
+                *d = strv_free(*d);
+
+        t->name = mfree(t->name);
+}
+
 UnitTimes* unit_times_free_array(UnitTimes *t) {
         if (!t)
                 return NULL;
 
         for (UnitTimes *p = t; p->has_data; p++)
-                free(p->name);
+                unit_times_clear(p);
 
         return mfree(t);
 }
 
+DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(UnitTimes*, unit_times_clear, NULL);
+
+#define UNIT_TIMES_DEP_FIELD(f) (offsetof(UnitTimes, deps) + (f * sizeof(char **)))
+
 int acquire_time_data(sd_bus *bus, bool require_finished, UnitTimes **out) {
         static const struct bus_properties_map property_map[] = {
-                { "InactiveExitTimestampMonotonic",  "t", NULL, offsetof(UnitTimes, activating)   },
-                { "ActiveEnterTimestampMonotonic",   "t", NULL, offsetof(UnitTimes, activated)    },
-                { "ActiveExitTimestampMonotonic",    "t", NULL, offsetof(UnitTimes, deactivating) },
-                { "InactiveEnterTimestampMonotonic", "t", NULL, offsetof(UnitTimes, deactivated)  },
+                { "InactiveExitTimestampMonotonic",  "t",  NULL, offsetof(UnitTimes, activating) },
+                { "ActiveEnterTimestampMonotonic",   "t",  NULL, offsetof(UnitTimes, activated) },
+                { "ActiveExitTimestampMonotonic",    "t",  NULL, offsetof(UnitTimes, deactivating) },
+                { "InactiveEnterTimestampMonotonic", "t",  NULL, offsetof(UnitTimes, deactivated) },
+                { "After",                           "as", NULL, UNIT_TIMES_DEP_FIELD(UNIT_AFTER) },
```

Comment:
Please simply use `offsetof(UnitTimes, deps[UNIT_AFTER])`.

Reply:
I had no idea that worked, that's really neat.